src/riak_kv_index_hashtree.erl

```diff
@@ -296,11 +302,12 @@ handle_call({delete, Items}, _From, State) ->
 handle_call(get_trees, _From, #state{trees=Trees}=State) ->
     {reply, Trees, State};
 
-handle_call({update_tree, Id}, From, State) ->
+handle_call({update_tree, Id, Callback}, From, State) ->
```

Comment:
For the one w/ the Callback.. add a guard -> `is_function(Callback)` or == undefined. Dialyze this handle_call specifically.